### PR TITLE
Add sqlboiler adapter to postgres.

### PR DESCRIPTION
**Pull request by ZaraGiraffe**
#21

### CODE REVIEW

File: core/pkg/core/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+func (s *Storage) GetStatus(id uuid.UUID) modules.DetectionStatus {
+	status, err := models.FindDetectionStatus(context.TODO(), s.db, id.String())
+	switch err {
+	case nil:
+		// Do nothing
+	case sql.ErrNoRows:
+		return modules.DetectionStatus{RequestId: id, Status: "UNKNOWN",}
+	default:
+		panic(fmt.Sprintf("Error in FindDetectionStatus, error: %s", err.Error()))
```
Comment (anakib1) [inline core/pkg/core/storage/storage.go:53]: Don't think, we need to panic. SQL could be down, let's return unknown status and write error to log

File: core/pkg/core/storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type Storage struct {
+	db boil.ContextExecutor
+}
+
+func NewStorage() *Storage {
+	db, err := sql.Open("postgres", "dbname=mango_truth host=localhost user=zahar password=password sslmode=disable")
```
Comment (anakib1) [inline core/pkg/core/storage/storage.go:21]: move to config

File: core/pkg/core/storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	if err != nil {
+		panic(fmt.Sprintf("Can not connect verdict to json in DetectionStatus with requestId=%s, error: %s", status.RequestId, err.Error()))
+	}
+	new_status := models.DetectionStatus{
+		RequestID: status.RequestId.String(),
+		Status: status.Status,
+		Data: null.BytesFrom(data),
+	}
+
+	err = new_status.Insert(context.TODO(), s.db, boil.Infer())
```
Comment (anakib1) [inline core/pkg/core/storage/storage.go:39]: I think, deadlines do more harm than good here, so we can pass background context (and initialise it as Storage.context) all the time

File: core/pkg/core/storage/schema.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE detection_status (
+    request_id VARCHAR(255) PRIMARY KEY,
+    status VARCHAR(255) NOT NULL,
```
Comment (anakib1) [inline core/pkg/core/storage/schema.sql:3]: Need to decide what to do with status. I would prefer it to be enum with mapping to int, not varchar

File: core/pkg/core/storage/schema.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE detection_status (
+    request_id VARCHAR(255) PRIMARY KEY,
```
Comment (anakib1) [inline core/pkg/core/storage/schema.sql:2]: uuid

File: core/pkg/core/storage/storage.go (hunk 3 above)
Comment (ZaraGiraffe) [reply core/pkg/core/storage/storage.go:41]: Why? context.TODO() does nothing. It is used only as a placeholder

File: core/pkg/core/storage/storage.go (hunk 1 above)
Comment (anakib1) [reply core/pkg/core/storage/storage.go]: @ZaraGiraffe

File: core/pkg/core/storage/schema.sql (hunk 4 above)
Comment (anakib1) [reply core/pkg/core/storage/schema.sql]: Let's try to do this way 

https://github.com/volatiletech/sqlboiler?tab=readme-ov-file#enums

File: core/pkg/core/storage/storage.go (hunk 1 above)
Comment (ZaraGiraffe) [reply core/pkg/core/storage/storage.go]: I have fixed it. I do not know why it is outdated. See [here](https://github.com/anakib1/MangoTruth/blob/add-sqlboiler-adapter/core/pkg/storage/storage.go)

File: core/pkg/storage/docker/schema.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+CREATE TYPE STATUS AS ENUM('UNKNOWN', 'PENDING', 'PARSING_FAILED', 'TRANSPORT_FAILED');
```
Comment (anakib1) [inline core/pkg/storage/docker/schema.sql:1]: You forgot `SUCCESS`  😄 

Also, let's have only one failure enum. Details could be stored as any text later. 


PENDING - in progress of detection, no status yet, but acknowledged 
UNKNOWN - request is completely unknown to server
SUCCESS - request was successfully processed and contains some verdict from compute 
FAILURE - anything went wrong during request processing on backend side.

File: core/pkg/storage/storage.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	new_status := models.DetectionStatus{
+		RequestID: status.RequestId.String(),
+		Status: status.Status,
+		Data: null.BytesFrom(data),
+	}
+
+	err = new_status.Insert(context.TODO(), s.db, boil.Infer())
+	if err != nil {
+		panic(fmt.Sprintf("Can not insert value into storage, error: %s", err.Error()))
```
Comment (anakib1) [inline core/pkg/storage/storage.go:45]: We should panic less. SQL could be down, but application can still function. Let's panic on something that actually should always happen - like json marshalling fails. That's actually bad.

File: core/pkg/storage/storage.go [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+	switch err {
+	case nil:
+		// Do nothing
+	case sql.ErrNoRows:
+		return modules.DetectionStatus{RequestId: id, Status: models.StatusUNKNOWN,}
+	default:
+		slog.Error("Error in FindDetectionStatus", "error-msg", err.Error())
+		return modules.DetectionStatus{RequestId: id, Status: models.StatusUNKNOWN,}
+	}
+	request_id, err := uuid.Parse(status.RequestID)
```
Comment (anakib1) [inline core/pkg/storage/storage.go:60]: MustParse

**Review by anakib1 — APPROVED** [review]

**Pull request merged**